larreco/RecoAlg/PMAlg/PmaElement3D.cxx

```diff
@@ -0,0 +1,342 @@
+/**
+ *  @file   PmaElement3D.cxx
+ *
+ *  @author D.Stefan and R.Sulej
+ * 
+ *  @brief  Implementation of the Projection Matching Algorithm
+ *
+ *          Base for 3D segments and track nodes. See PmaTrack3D.h file for details.
+ *          See PmaTrack3D.h file for details.
+ */
+
+#include "larreco/RecoAlg/PMAlg/PmaElement3D.h"
+#include "larreco/RecoAlg/PMAlg/SortedObjects.h"
+#include "larreco/RecoAlg/PMAlg/Utilities.h"
+
+#include "messagefacility/MessageLogger/MessageLogger.h"
+
+// Impact factors on the objective function:  U     V     Z
+float pma::Element3D::fOptFactors[3] =     { 0.2F, 0.8F, 1.0F };
+
+pma::Element3D::Element3D() :
+	fTPC(-1), fCryo(-1),
```

Comment:
I'd suggest that instead of using `-1` explicitly, create symbolic constants,  e.g. `kInvalidTPC` and `kInvalidCryo`, which you set to `-1`. Then user code could use those values and it would help readability of the code. In addition, it would be possible to change the representation of the value (e.g. become `unsigned`) and using symbolic names would keep code working.